session/session.go

```diff
@@ -1272,6 +1272,34 @@ const loadCommonGlobalVarsSQL = "select HIGH_PRIORITY * from mysql.global_variab
 	variable.TiDBHashJoinConcurrency + quoteCommaQuote +
 	variable.TiDBDistSQLScanConcurrency + "')"
 
+type globalVariableCache struct {
+	sync.RWMutex
+	lastModify time.Time
+	rows       []types.Row
+	fields     []*ast.ResultField
+}
+
+var gvc globalVariableCache
+
+func (gvc *globalVariableCache) Update(rows []types.Row, fields []*ast.ResultField) {
+	gvc.Lock()
+	gvc.lastModify = time.Now()
+	gvc.rows = rows
+	gvc.fields = fields
+	gvc.Unlock()
+}
+
+func (gvc *globalVariableCache) Get() (succ bool, rows []types.Row, fields []*ast.ResultField) {
+	gvc.RLock()
+	defer gvc.RUnlock()
+	if time.Now().Sub(gvc.lastModify) < 2*time.Second {
```

Comment:
Make it constant.
And I think 2 seconds is too small, how about 10 seconds?

Reply:
10 seconds is too long!
2 seconds maybe a good balance between correctness and performance.